printer/printast_with_mappings.ml

```diff
@@ -159,11 +169,13 @@ let rec core_type i ppf x =
   match x.ptyp_desc with
   | Ptyp_any -> line i ppf "Ptyp_any\n";
   | Ptyp_var (s) -> line i ppf "Ptyp_var %s\n" s;
-  | Ptyp_arrow (l, ct1, ct2) ->
+  | Ptyp_arrow (l, ct1, ct2, m1, m2) ->
       line i ppf "Ptyp_arrow\n";
       arg_label i ppf l;
       core_type i ppf ct1;
       core_type i ppf ct2;
+      modes i ppf m1;
+      modes i ppf m2;
```

Comment:
The two lists of modes might be indistinguishable. Might be better if we print `ct1`, `m1`, then `ct2` and `m2`.